VerilogCodingStyle.md

```diff
@@ -2539,6 +2539,52 @@ function automatic logic [2:0] foo(logic [2:0] a, logic [2:0] b);
 endfunction
 ```
 
+Functions should not reference any non-local signals, variables, or memories
```

Comment:
I would leave out "memories" because it's really just a variable that is an unpacked array.